File: src/backend/tests/c-lisp/kernel/runtime/MMult2.c
#include "add_dot.c"

void kernel(float* a, float* b, float* c, int m, int n, int k){
    
    int lda = m;
    int ldb = n;
    int ldc = k;

    for(int j = 0; j < n; j += 4) {
        for(int i = 0; i < m; i++) {
            add_dot(k, &a[0 * lda + i], lda, &b[(j + 0) * ldb], &c[(j + 0) * ldc + i]);
            add_dot(k, &a[0 * lda + i], lda, &b[(j + 1) * ldb], &c[(j + 1) * ldc + i]);
            add_dot(k, &a[0 * lda + i], lda, &b[(j + 2) * ldb], &c[(j + 2) * ldc + i]);
            add_dot(k, &a[0 * lda + i], lda, &b[(j + 3) * ldb], &c[(j + 3) * ldc + i]);
        }
    }
}

File: src/backend/tests/c-lisp/kernel/kernels_c-lisp/MMult2.py
def call_add_dot(a, b, c, lda, ldb, ldc, ele, i, j, k):
    return [
        ('call', '__add_dot', 
            k, 
            ('ptradd', a, ('add', i, ('mul', 0, lda))), 
            lda,
            ('ptradd', b, ('add', 0, ('mul', ('add', j, ele), ldb))),
            ('ptradd', c, ('add', i, ('mul', ('add', j, ele), ldc))))
    ]


